core/src/main/scala/sttp/tapir/server/interpreter/ServerInterpreter.scala

```diff
@@ -84,10 +84,12 @@ class ServerInterpreter[R, F[_]: MonadError, B, S](
       case values: DecodeBasicInputsResult.Values =>
         values.bodyInputWithIndex match {
           case Some((Left(bodyInput @ EndpointIO.Body(_, codec, _)), _)) =>
-            requestBody.toRaw(bodyInput.bodyType).map { v =>
-              codec.decode(v) match {
-                case DecodeResult.Value(bodyV)     => values.setBodyInputValue(bodyV)
-                case failure: DecodeResult.Failure => DecodeBasicInputsResult.Failure(bodyInput, failure): DecodeBasicInputsResult
+            requestBody.toRaw(bodyInput.bodyType).flatMap { v =>
+              codec.decode(v.value) match {
+                case DecodeResult.Value(bodyV) => (values.setBodyInputValue(bodyV): DecodeBasicInputsResult).unit
+                case failure: DecodeResult.Failure =>
+                  m.eval(v.tmpFiles.foreach(_.delete()))
```

Comment:
I think just as we parametrise the interpreters with a way to create the files (via `XOptions`), we should have the same for deleting files. It's a blocking operation so it might need to run on a separate thread pool, for example.